lib/board.rb

```diff
@@ -0,0 +1,52 @@
+class Board
+  attr_accessor :cells
+
+
+  def initialize
+    self.reset!
+  end
+
+  def reset!
+    @cells = Array.new(9," ")
+  end
+
+  def position
+    x = gets.chomp.to_i
+    x-=1
+    @cells[x]
+  end
+
+  def display
+    puts " #{@cells[0]} | #{@cells[1]} | #{@cells[2]} \n-----------\n #{@cells[3]} | #{@cells[4]} | #{@cells[5]} \n-----------\n #{@cells[6]} | #{@cells[7]} | #{@cells[8]} "
+  end
+  #to display before the start of the game, making it user-friendly
+  def display_intro
+    puts " 1 | 2 | 3 \n-----------\n 4 | 5 | 6 \n-----------\n 7 | 8 | 9 "
+  end
+
+  def full?
+    !@cells.any?{|x| x == " " || x == "" }
```

Comment:
Use the most expressive and efficient iterator for a problem. any? is a pretty good choice, but I think we can do better. Board#full? is asking if all elements are not empty strings. You can use logic that is symmetrical to those semantics with the #all? iterator.

@cells.all? {|x| x != " " && x != ""}

Sometimes, having a method that returns true for a certain situation is more readable than using ! with a method that returns false.